### PR TITLE
feat(datetime-picker): cascade custom cell slot to calendar

### PR DESCRIPTION
**Pull request by bualoy-napat**
## Description

As datetime-picker attaches calendar inside, we also need to cascade the custom cell slot on from datetime-picker to calendar.

Fixes # (issue)
ELF-2267

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: documents/src/pages/elements/datetime-picker.md [hunk 1]
```diff
@@ -343,6 +343,259 @@ section {
 </ef-datetime-picker>
 ```
 
+## Custom cells
+Datetime-picker allows you to customise cells on the calendar using slots. Slot name could begin with either `from-` or `to-` as a prefix, indicating which side of the calendar to be customised. After the prefix, you need to specify which date, month, or year to be customised with `yyyy-MM-dd`, `yyyy-MM`, or `yyyy`. If the prefix is not defined, the slot will apply to "from" calendar only.
```
Comment (wattachai-lseg) [inline documents/src/pages/elements/datetime-picker.md]:
```suggestion
Datetime Picker allows you to customise cells on the calendar using slots. Slot name could begin with either `from-` or `to-` as a prefix, indicating which side of the calendar to be customised. After the prefix, you need to specify which date, month, or year to be customised with `yyyy-MM-dd`, `yyyy-MM`, or `yyyy`. If the prefix is not defined, the slot will apply to "from" calendar only.
```

**Review by wattachai-lseg — COMMENTED** [review]
only one tiny change.

**Comment by wattachai-lseg** [comment]
LGTM. Gonna approve this once it's been verified.

File: documents/src/pages/elements/datetime-picker.md [hunk 2]
```diff
@@ -343,6 +343,259 @@ section {
...
+      28: GDP Growth Rate YoY Flash
+    </p>
+    <h6>May</h6>
+    <p>
+      16: ZEW Economic Sentiment Index<br>
+      25: GfK Consumer Confidence<br>
+      31: Inflation Rate YoY Prel
+    </p>
+  </div>
+  <ef-datetime-picker opened duplex range lang="de" lang="de">
```
Comment (Sakchai-Refinitiv) [inline documents/src/pages/elements/datetime-picker.md]:
```suggestion
  <ef-datetime-picker opened duplex range lang="de">
```

File: documents/src/pages/elements/datetime-picker.md [hunk 3]
```diff
@@ -343,6 +343,259 @@ section {
...
+  <div class="custom-cell" slot="2023-04"></div>
+  <div class="custom-cell" slot="2023-05"></div>
+  <div class="custom-cell" slot="2023"></div>
+</ef-calendar>
+```
+
+```javascript
+const datetimePicker = document.querySelector('ef-datetime-picker');
+
+datetimePicker?.addEventListener('before-cell-render', (event) => {
```
Comment (Sakchai-Refinitiv) [inline documents/src/pages/elements/datetime-picker.md]:
```suggestion
datetimePicker.addEventListener('before-cell-render', (event) => {
```

File: documents/src/pages/elements/datetime-picker.md (hunk 2 above)
Comment (bualoy-napat) [reply documents/src/pages/elements/datetime-picker.md]: fixed

File: documents/src/pages/elements/datetime-picker.md (hunk 3 above)
Comment (bualoy-napat) [reply documents/src/pages/elements/datetime-picker.md]: fixed

**Comment by Sarin-Udompanish** [comment]
LGTM.

**Review by wattachai-lseg — APPROVED** [review]

**Pull request merged**